src/go/configmanager/flags/flags.go

```diff
@@ -144,7 +144,7 @@ var (
 	JwksFetchRetryBackOffMaxIntervalMs  = flag.Int("jwks_fetch_retry_back_off_max_interval_ms", int(defaults.JwksFetchRetryBackOffMaxInterval.Milliseconds()), `Specify JWKS fetch retry exponential back off maximum interval in milliseconds. The default is 32 seconds.`)
 	JwtPatForwardPayloadHeader          = flag.Bool("jwt_pad_forward_payload_header", defaults.JwtPadForwardPayloadHeader, `For the JWT in request, the JWT payload is forwarded to backend in the "X-Endpoint-API-UserInfo"" header by default. 
 Normally JWT based64 encode doesn’t add padding. If this flag is true, the header will be padded.`)
-	JwtCacheSize = flag.Uint("jwt_cache_size", defaults.JwtCacheSize, `Specify JWT cache size, the number of unique JWT tokens in the cache. The cache only stores verified good tokens. If 0, JWT cache is disabled. The default is 0.`)
+	JwtCacheSize = flag.Uint("jwt_cache_size", defaults.JwtCacheSize, `Specify JWT cache size, the number of unique JWT tokens in the cache. The cache only stores verified good tokens. If 0, JWT cache is disabled. The default is 1000.`)
```

Comment:
Curious why you choose 1000. This seems pretty small for 1 whole hour TBH.

Reply:
it is the number of unique tokens in 1 hour.      My gut feeling is it should be small, e.g. SA tokens

For end user tokens,  it is too small.  

It limits the memory we allocated for the cache.   Normally 

a token is  100B,  or 0.1K,  
1k tokens is 0.1M.   
100k tokens? which is 10M

I guess we can set the cache size as 1m,  which is only 100M memory.

What do you think?

Reply:
Let's do 100k tokens or 10MB.

Based on Cloud ESF benchmark report, baseline memory usage is 200 MB. We can add an extra 5% for cache. 100M extra is 50%, too much.

(I am assuming ESPv2 has similar performance)

Reply:
Also I don't know how the cache is implemented internally, but it may have extra memory usage per item for LRU or tracking. With this assumption 100k tokens or 10MB makes sense still

Reply:
@nareddyt @qiwzhang https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/jwt_authn/jwt_cache.cc#L25-L27 shows that config with `jwt_cache_size` set to `0` is not disabled (as flag description states here) but uses envoy's default cache size `100`